File: app/src/main/java/in/sudopk/vaishnavacalendar/Location.java
package in.sudopk.vaishnavacalendar;

import android.support.annotation.NonNull;

public class Location implements Comparable<Location> {
    private String mName;
    private String mId;

    public Location(final String name, final String id) {
        mName = name;
        mId = id;
    }

    public String getId() {
        return mId;
    }

    public String getName() {
        return mName;
    }

    @Override
    public int compareTo(@NonNull final Location o) {
        return getName().compareTo(o.getName());
    }

    @Override
    public boolean equals(final Object obj) {
        return (obj instanceof Location) && compareTo((Location) obj) == 0;
    }
}

File: app/src/main/java/com/sudopk/vaishnavacalendar/location/Location.kt
package com.sudopk.vaishnavacalendar.location

class Location(val name: String, val id: String) : Comparable<Location> {

    override fun compareTo(other: Location): Int {
        return name.compareTo(other.name)
    }

    override fun equals(other: Any?): Boolean {
        return other is Location && compareTo((other)) == 0
    }

    override fun hashCode(): Int {
        return name.hashCode();
    }
}
